packages/vite/src/node/utils.ts

```diff
@@ -282,21 +282,8 @@ export const isJSRequest = (url: string): boolean => {
   return false
 }
 
-const knownTsRE = /\.(?:ts|mts|cts|tsx)$/
-const knownTsOutputRE = /\.(?:js|mjs|cjs|jsx)$/
+const knownTsRE = /\.(?:ts|mts|cts|tsx)(?:$|\?)/
```

Comment:
FYI This `perf` PR also fixes a bug we run into in Vitest. I was just about to set up a fix PR for it but this exact line fixed it - thanks! My changes for this regexp would have been identical with this one.
Here's minimal repro for it just in case you want to set a test case to prevent this from breaking in future:

```js
import { writeFileSync } from "fs";
import { fileURLToPath } from "url";
import { createServer } from "vite";

writeFileSync(
  "./source.ts",
  `import { hello } from "./target.mjs"; // Change this to ".mts" or omit extension and it works too
  hello();`,
  "utf8"
);
writeFileSync(
  "./target.mts",
  `export const hello = () => "Hello from .mts";`,
  "utf8"
);

const __dirname = fileURLToPath(new URL(".", import.meta.url));

const server = await createServer({
  configFile: false,
  root: __dirname,
  server: {
    port: 1337,
  },
});
await server.listen();

const first = await server.transformRequest("./source.ts");
console.log("First worked!", first.code);

await new Promise((resolve) => setTimeout(resolve, 2000));

const second = await server
  .transformRequest("./source.ts?v=123")
  .then((result) => console.log("Second worked!", result.code))
  .catch((e) => console.error("Second failed", e));

await server.close();
process.exit();
```

Reply:
Awesome! Thanks for letting us know! Maybe not a bad idea to send a PR to add a case for this to one of our playgrounds if you would like.